platform/mellanox/integration-scripts/hwmgmt_kernel_patches.py

```diff
@@ -0,0 +1,314 @@
+                if index_prev_patch < len(Data.non_up_slk_series) - 1:
+                    Data.non_up_slk_series = Data.non_up_slk_series[0:index_prev_patch+1] + [patch] +  Data.non_up_slk_series[index_prev_patch + 1:]
+                else:
+                    Data.non_up_slk_series = Data.non_up_slk_series + [patch]
+                lines.append(patch.strip())
+
+        # Update the non_up_current_patch_list file
+        FileHandler.write_lines(self.args.current_non_up_patches, lines)
+        print("\n -> POST: series file updated with non-upstream patches \n{}".format("".join(Data.non_up_slk_series)))
+
+    def write_series_diff(self):
+        diff = difflib.unified_diff(Data.final_slk_series, Data.non_up_slk_series, fromfile='a/patch/series', tofile="b/patch/series", lineterm="\n")
+        lines = []
+        for line in diff:
+            lines.append(line)
+        print("\n -> POST: final series.diff \n{}".format("".join(lines)))
+        FileHandler.write_lines(os.path.join(self.args.build_root, NON_UP_PATCH_DIFF), lines, True)
+    
+    def find_conflicts(self):
+        # current config under mellanox marker
+        old_mlnx_kcfg =  FileHandler.read_kconfig_inclusion(os.path.join(self.args.build_root, SLK_KCONFIG))
+        old_mlnx_kcfg = KCFG.parse_opts_strs(old_mlnx_kcfg)
+
+        print("-> INFO: [common] + [amd64] Kconfig: \n{}".format("\n".join(KCFG.get_writable_opts(Data.current_kcfg))))
+        print("-> INFO: current mellanox marker Kconfig: \n{}".format("\n".join(KCFG.get_writable_opts(old_mlnx_kcfg))))
+
+        # Filter the mellanox config from current config
+        conflict_prone = set(Data.current_kcfg)
+        for kcfg in old_mlnx_kcfg:
+            if kcfg in conflict_prone:
+                conflict_prone.remove(kcfg)
+
+        print("-> INFO: conflict prone Kconfig: \n{}".format("\n".join(KCFG.get_writable_opts(list(conflict_prone)))))
+        print("-> INFO: updated kconfig for mellanox marker: \n{}".format("\n".join(KCFG.get_writable_opts(Data.updated_kcfg))))
+
+        # check for conflicts
+        has_conflict = False
+        for (cfg, val) in Data.updated_kcfg:
+            for (cfg_o, val_o) in conflict_prone:
+                if cfg == cfg_o and val != val_o:
+                    print("-> ERR Conflict seen on the following kconfig: {}, old_opt: {}, new_opt: {}".format(cfg, val_o, val))
+                    has_conflict = True
+        return has_conflict
+
+    def handle_exclusions(self):
+        new_lines = []
+        curr_hdr = ""
+        for line_raw in Data.kcfg_exclude:
+            line = line_raw.strip()
+            should_exclude = False
+            if line_raw.strip():
+                match = re.search(KCFG_HDR_RE, line)
+                if match:
+                    curr_hdr = match.group(1)
+                else:
+                    for (kcfg, _) in Data.updated_kcfg:
+                        if kcfg == line and curr_hdr in HDRS:
```

Comment:
The current design has limitation that hw-mgmt provided configs belong to amd64 and common. Hence the script needs to be revisited when support for arm64 is introduced

Reply:
Yes, have to check with hw-mgmt team.